### DISCUSSION TITLE
vSphere Web Services API vs. vSphere Automation API

### DISCUSSION THREAD

**Discussion by jirihnidek** [vmware/govmomi#3580] [Community Help]
Hi,
I know that `govmomi` uses "vSphere Web Services API":

https://developer.broadcom.com/xapis/vsphere-web-services-api/latest/

There is also another (newer?) REST API called "vSphere Automation API":

https://developer.broadcom.com/xapis/vsphere-automation-api/latest/

It seems that intentions of these two APIs are little bit overlapping. Is there any risk that REST API will have higher priority over SOAP based API and SOAP based API will be deprecated? 

Thanks in advance,

Jiri

**Comment by dougm** [comment]
govmomi actually uses both SOAP (vmodl1) and REST (vmodl2) APIs. Majority is SOAP, but newer features are only available in the REST APIs - such as tagging, content library and namespaces. There is some overlap between the two, but vmodl1/SOAP won't go away any time soon. @karaatanassov may have some more insights to share.

**Comment by jirihnidek** [comment]
Thanks for information. It is much appreciated. :+1:
